a3p-integration/proposals/a:upgrade-next/priceFeed.test.js

```diff
@@ -0,0 +1,155 @@
+/* global process */
+
+import test from 'ava';
+
+import {
+  agd,
+  agops,
+  executeOffer,
+  getVatDetails,
+  GOV1ADDR,
+  GOV2ADDR,
+  GOV3ADDR,
+  newOfferId,
+} from '@agoric/synthetic-chain';
+
+const ORACLE_ADDRESSES = [GOV1ADDR, GOV2ADDR, GOV3ADDR];
+
+const getOracleInstance = async price => {
+  const instanceRec = await agd.query(
+    'vstorage',
+    'data',
+    '--output',
+    'json',
+    `published.agoricNames.instance`,
+  );
+
+  // agd query -o json  vstorage data published.agoricNames.instance
+  //    |& jq '.value | fromjson | .values[1] | fromjson | .body[1:]
+  //    | fromjson | .[-2] '
+
+  const value = JSON.parse(instanceRec.value);
+  // XXX I don't know why value.values[-1] doesn't work.
+  const body = JSON.parse(value.values[value.values.length - 1]);
```

Comment:
because JS arrays are also dicts and `-1` is a valid key

```suggestion
  const body = JSON.parse(value.values.at(-1));
```

Reply:
thanks.